app/src/main/java/com/example/bottomnav/Parse.java

```diff
@@ -71,6 +76,51 @@ private void storeStuct(IASTDeclaration[] declarations, String name) throws Exce
         structRepo.put(name, struct);
     }
 
+    public void mapPayload(IASTNode[] comments) {
+
+        for (IASTNode comment : comments) {
+            char[] line = simplifyString(comment.getRawSignature());
+            if (line[0] == '|') {
+                String idName = findString(line, 1);
+                int secondIndex = idName.length() + 2;
+                String structName = findString(line, secondIndex);
+                if (!idStruct.containsKey(structName)) {
+                    PayloadMap rep = new PayloadMap(structName);
+                    idStruct.put(structName, rep);
+                }
+                idStruct.get(structName).canIDNames.add(idName);
+            }
+        }
+    }
+
+    public char[] simplifyString(String line) {
+        line = line.replace("//", "");
+        line = line.replace(" ", "");
+        return line.toCharArray();
+    }
+
+    public String findString(char[] line, int start){
+        StringBuilder name = new StringBuilder();
+        char character = line[start];
+        while (character != '|') {
+            name.append(character);
+            start += 1;
+            character = line[start];
+        }
+        return name.toString();
+    }
+
+    //Given a CAN ID name, retrieves its associated struct
+    public ArrayList getAssociatedStruct(String idName) {
```

Comment:
Why does this return an `ArrayList`? Is there ever a case where you can return more than one struct for a given name? Would `Optional<StructContents>` be a better fit?

Similarly for `getStruct` - maybe that needs to be refactored?